server/monitor.go

```diff
@@ -2211,3 +2211,264 @@ func (s *Server) accountInfo(accName string) (*AccountInfo, error) {
 		responses,
 	}, nil
 }
+
+// LeafzOptions are options passed to Jsz
+type JSzOptions struct {
+	Account  string `json:"account,omitempty"`
+	Accounts bool   `json:"accounts,omitempty"`
+	Streams  bool   `json:"streams,omitempty"`
+	Consumer bool   `json:"consumer,omitempty"`
+	Offset   int    `json:"offset,omitempty"`
+	Limit    int    `json:"limit,omitempty"`
+}
+
+type StreamDetail struct {
+	Name     string          `json:"name"`
+	Cluster  *ClusterInfo    `json:"cluster,omitempty"`
+	Config   StreamConfig    `json:"config,omitempty"`
```

Comment:
probably we can drop the config or make it optional, it's quite verbose and we'd need to handle huge amounts of them in theory, ditto consumer config

Reply:
made it optional for streams and consumer.
I kept the one of jetstream itself as its small, only included once and helps put memory/storage numbers into context.